File: flink-scala-core/src/main/scala/com/atguigu/flink/day2/SensorSource.scala
package com.atguigu.flink.day2

import org.apache.flink.streaming.api.functions.source.SourceFunction.SourceContext
import org.apache.flink.streaming.api.functions.source.RichParallelSourceFunction

import scala.util.Random

class SensorSource extends RichParallelSourceFunction[SensorReading]{

    var running = true

    override def run(ctx: SourceContext[SensorReading]): Unit = {
        val rand = new Random
        val curFTemp = (1 to 10).map(i=>("Sensor_"+i,rand.nextGaussian()*20))
        while(true){
            curFTemp.foreach{
                r =>{
                    ctx.collect(new SensorReading(r._1,System.currentTimeMillis(),r._2 + rand.nextGaussian() * 5))
                }
            }
            Thread.sleep(300)
        }
    }

    override def cancel(): Unit = {
        running = false
    }
}

File: flink-java-core/src/main/java/com/atguigu/flink/day2/SensorSource.java
package com.atguigu.flink.day2;

import org.apache.flink.streaming.api.functions.source.RichParallelSourceFunction;

import java.util.Random;

/**
 * @author ：hmz
 * @date ：Created in 2020/9/11 10:05
 */
public class SensorSource extends RichParallelSourceFunction<SensorReading> {

    private boolean running = true;
    private Random random = new Random();

    @Override
    public void run(SourceContext<SensorReading> ctx) throws Exception {
        while (true) {
            for (int i = 1; i <= 10; i++) {
                ctx.collect(new SensorReading("Sensor_" + i, System.currentTimeMillis(), getRandomTemperature()));
            }
            Thread.sleep(300);
        }
    }

    @Override
    public void cancel() {
        running = false;
    }

    private Double getRandomTemperature() {
        return random.nextGaussian() * 20 + random.nextGaussian() * 5;
    }
}
